lz4.go

```diff
@@ -334,15 +336,27 @@ type CompressReader struct {
 // in the lz4 library will not be freed. The compressed output must be decompressed
 // using NewDecompressReader.
 func NewCompressReader(r io.Reader) *CompressReader {
+	// The input buffers MUST NOT be contiguous in memory so the two blocks are treated as separate.
+	// We had a bug in Writer when malloc decided to allocate buffers contiguously. This bug does
+	// not happen with CompressReader, because we only have "partial" blocks at EOF, and we need two
+	// calls to LZ4_compress_fast_continue with sizes < 64 kiB to trigger the problem. However, we
+	// should separate these buffers explicitly, to make this impossible. For details, see the
+	// comment in NewWriter.
+
+	// separate the buffers by 8 bytes, so LZ4 treats them as separate. 8 bytes means the buffer
+	// start is 8 byte aligned, which may permit optimizations on 64-bit CPUs.
+	const bufferSeparation = 8
```

Comment:
That's not accurate. Adding `8` to a pointer doesn't make it aligned. The pointer needs to be aligned beforehand for this to hold. If you really want to align you'd have to do

```
ptr = (ptr + align - 1) & !(ptr - 1)
```

I don't think it matters much here but I thought I'd mention it. There's also a chance I'm just misreading all of it

Reply:
Hum, I now see that the alignment comment was copy-pasta. Maybe just remove the mention to alignment :shrug:

Reply:
And please don't git blame the comment itself lol I think it's me circa 2019

Reply:
I'll edit the comment in both places, because it is wrong, or at least misleading, thanks! My assumption is that the pointer returned by malloc will be 8 byte aligned on a 64 bit machine, due to the requirement that "The allocated memory is aligned such that it can be used for any data type." There are some incredible details here: https://github.com/jemalloc/jemalloc/issues/1533 TL;DR: It seems like the block is guaranteed to 16 byte aligned, so adding 8 will do the right thing. Still: the comment is definitely wrong. Thanks!